File: app/src/main/java/pl/piotrskiba/teatime/utils/LanguageUtils.java
package pl.piotrskiba.teatime.utils;

import android.content.Context;
import android.content.SharedPreferences;
import android.content.res.Configuration;
import android.content.res.Resources;
import android.preference.PreferenceManager;

import java.util.Locale;

import pl.piotrskiba.teatime.R;

public class LanguageUtils {
    public static void setAppLanguage(Context baseContext, String lang_code){
        // update language
        Locale locale = new Locale(lang_code);
        Locale.setDefault(locale);
        Configuration config = new Configuration();
        config.locale = locale;
        baseContext.getResources().updateConfiguration(config,
                baseContext.getResources().getDisplayMetrics());
    }

    public static String getLanguageFromSettings(Context context){
        SharedPreferences sharedPreferences = PreferenceManager.getDefaultSharedPreferences(context);
        String language = sharedPreferences.getString(context.getString(R.string.pref_language_key), getPhoneLanguage());

        if(language.equals(context.getString(R.string.pref_language_value_default)))
            return getPhoneLanguage();
        else
            return language;
    }

    public static String getAppLanguage(){
        return Locale.getDefault().getLanguage();
    }

    public static String getPhoneLanguage(){
        Locale globalLocale = Resources.getSystem().getConfiguration().locale;
        return globalLocale.getLanguage();
    }
}

File: app/src/main/java/pl/piotrskiba/teatime/utils/LanguageUtils.kt
package pl.piotrskiba.teatime.utils

import android.content.Context
import android.content.res.Configuration
import android.content.res.Resources
import androidx.preference.PreferenceManager
import pl.piotrskiba.teatime.R
import java.util.*

object LanguageUtils {
    @JvmStatic
    fun setAppLanguage(baseContext: Context, lang_code: String) { // update language
        val locale = Locale(lang_code)
        Locale.setDefault(locale)

        val config = Configuration()
        config.locale = locale

        baseContext.resources.updateConfiguration(config,
                baseContext.resources.displayMetrics)
    }

    @JvmStatic
    fun getLanguageFromSettings(context: Context): String? {
        val sharedPreferences = PreferenceManager.getDefaultSharedPreferences(context)

        val language = sharedPreferences.getString(context.getString(R.string.pref_language_key), phoneLanguage)
        return if (language == context.getString(R.string.pref_language_value_default)) phoneLanguage else language
    }

    @JvmStatic
    val appLanguage: String
        get() = Locale.getDefault().language

    val phoneLanguage: String
        get() {
            val globalLocale = Resources.getSystem().configuration.locale
            return globalLocale.language
        }
}